### PR TITLE
option to disable testing by setting BUILD_TESTING to OFF

### PR DESCRIPTION
**Pull request by stigh**
Removed enable_testing(), since this seems to be enabled when the subdirectory "tests" is added.
Used the intrinsic CTests name BUILD_TESTING, ref: https://cmake.org/cmake/help/latest/module/CTest.html.

From my understanding, maybe the ideal way of tackling this issue would be to handle BUILD_TESTING as an intrinsic setting in test-drive? Then we might not need the
`option(BUILD_TESTING "Define if tests should be enabled" ON)`
in CMakeLists.txt

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -114,7 +114,9 @@ else()
   target_sources(${PROJECT_NAME} PRIVATE f08estop.f90)
 endif()
 
-add_subdirectory(tests)
+if(BUILD_TESTING)
+  add_subdirectory(tests)
```
Comment (awvwgk) [inline src/CMakeLists.txt:118]:
```suggestion
  enable_testing()
  add_subdirectory(tests)
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -12,7 +12,7 @@ list(GET VERSION_LIST 1 PROJECT_VERSION_MINOR)
 list(GET VERSION_LIST 2 PROJECT_VERSION_PATCH)
 unset(VERSION_LIST)
 
-enable_testing()
+option(BUILD_TESTING "Define if tests should be enabled" ON)
```
Comment (awvwgk) [inline CMakeLists.txt:15]:
```suggestion
include(CTest)
```

**Review by awvwgk — APPROVED** [review]
Thanks for sharing. Looks like a useful addition.

**Review by jvdp1 — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**